### PR TITLE
Issue #263 Make proxy local development ready

### LINKED ISSUE #263: Setup Proxy for development
This could be either SSL support https://github.com/fabric8-services/fabric8-jenkins-proxy/issues/174
Or using dev cluster, 
Do whatever is required. 
We shouldn't be needing prod-preview to test our changes

### PR DESCRIPTION
**Pull request by kishansagathiya**
Proxy runs only on secured host. Thus, for running proxy locally we
need to make localhost:8080 SSL enabled. This commit makes that process
easier.

Fixes #263

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @kishansagathiya for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


For more information please head over to official [documentation](http://arquillian.org/ike-prow-plugins/#_test_keeper_plugin). You can find there how to configure the plugin - for example exclude certain file types so if PR contains only them it won't be checked.

File: cmd/fabric8-jenkins-proxy/main.go [hunk 1]
```diff
@@ -111,6 +112,24 @@ func start(config configuration.Configuration, tenant *clients.Tenant, wit clien
 	wg.Wait()
 }
 
+func listenAndServeHelper(srv *http.Server, cancel context.CancelFunc) {
+	IsSslEnabledStr, _ := os.LookupEnv("JC_ENABLE_SSL")
+	bSslEnabled, parseErr := strconv.ParseBool(IsSslEnabledStr)
+	if parseErr == nil {
+		if bSslEnabled {
```
Comment (lordofthejars) [inline cmd/fabric8-jenkins-proxy/main.go]: naming variables with this nomenclature reminds me some dark ages hahahaha but well probably it is fine.

File: cmd/fabric8-jenkins-proxy/main.go [hunk 2]
```diff
@@ -111,6 +112,24 @@ func start(config configuration.Configuration, tenant *clients.Tenant, wit clien
 	wg.Wait()
 }
 
+func listenAndServeHelper(srv *http.Server, cancel context.CancelFunc) {
+	IsSslEnabledStr, _ := os.LookupEnv("JC_ENABLE_SSL")
+	bSslEnabled, parseErr := strconv.ParseBool(IsSslEnabledStr)
+	if parseErr == nil {
```
Comment (containscafeine) [inline cmd/fabric8-jenkins-proxy/main.go]: if `err != nil` is generally called right after the call that returns the error, it's probably not idiomatic to check `if err == nil`.

File: cmd/fabric8-jenkins-proxy/main.go [hunk 3]
```diff
@@ -111,6 +112,24 @@ func start(config configuration.Configuration, tenant *clients.Tenant, wit clien
...
+		if bSslEnabled {
+			if err := srv.ListenAndServeTLS("server.crt", "server.key"); err != nil {
+				cancel()
```
Comment (containscafeine) [inline cmd/fabric8-jenkins-proxy/main.go]: a log message here would help

File: cmd/fabric8-jenkins-proxy/main.go [hunk 4]
```diff
@@ -111,6 +112,24 @@ func start(config configuration.Configuration, tenant *clients.Tenant, wit clien
...
+				return
+			}
+		} else {
+			if err := srv.ListenAndServe(); err != nil {
+				cancel()
```
Comment (containscafeine) [inline cmd/fabric8-jenkins-proxy/main.go]: add log

File: cmd/fabric8-jenkins-proxy/main.go [hunk 5]
```diff
@@ -106,12 +106,30 @@ func start(config configuration.Configuration, tenant *clients.Tenant, wit clien
...
-	startWorkers(ctx, &wg, cancel, store, &proxy, defaultStatsLoggingInterval, config.GetDebugMode())
+	startWorkers(ctx, &wg, cancel, store, &proxy, defaultStatsLoggingInterval, config)
 	setupSignalChannel(cancel)
 	wg.Wait()
 }
 
-func startWorkers(ctx context.Context, wg *sync.WaitGroup, cancel context.CancelFunc, store storage.Store, proxy *proxy.Proxy, interval time.Duration, addProfiler bool) {
+func listenAndServe(srv *http.Server, cancel context.CancelFunc, enableHTTPS bool) {
+	if enableHTTPS {
+		if err := srv.ListenAndServeTLS("server.crt", "server.key"); err != nil {
```
Comment (sthaha) [inline cmd/fabric8-jenkins-proxy/main.go]: need to log error otherwise this fails silently

File: cmd/fabric8-jenkins-proxy/main.go [hunk 6]
```diff
@@ -106,12 +106,30 @@ func start(config configuration.Configuration, tenant *clients.Tenant, wit clien
...
+			cancel()
+			return
+		}
+	} else {
+		if err := srv.ListenAndServe(); err != nil {
```
Comment (sthaha) [inline cmd/fabric8-jenkins-proxy/main.go]: same here

**Comment by sthaha** [comment]
Merging this based on @kishansagathiya comment

**Pull request merged**